### ISSUE TITLE
Automatic hashsums for the downloads

### ISSUE THREAD

**Issue by XVilka** [radare/radare.org#29]
Especially for those from the CI server (like windows exe and zip)

**Comment by radare** [comment]
Checksums are useless. We already discussed this and decided to use signatures. So the installer can ship the pubkey and verify the download with it. I started using signify, but this is going to be a problem for android, where i would prefer an implementation in pure java

**Comment by jvoisin** [comment]
Why `signify` and not `gpg`?
It would be cool to document our thread model about this, and what motivated our decisions.

**Comment by radare** [comment]
Now that gpg fixed all the buffer overflows and infinite loops i found, gpg can be an option too. But have the same issues than signify for android. Because ill have to do a build of it for 4 different archs, and gpg is bigger.

Maybe plain rsa, so just using openssl, which can be done in java, objc.. But signify looks shorter and i was thinking in crosscompiling it to js and use the js vm to run that code on all platforms (android, ios).. And just use a native build of it for osx/w32/linux.

**Comment by jvoisin** [comment]
I'm quite sure that we don't want to implement our own crypto. I'd vote for gpg, because it's what everyone is using, or signify, since it's maintained, and likely audited.

**Comment by radare** [comment]
Yep. Thats why i suggested to use signify. Because its used by openbsd, which seems legit. Its codebase is smaller than gpg and we can even... Include signify inside the r2 codebase so we can just use r2 to verify r2. But this will depend on trusting the initial downloaded tarball.

**Comment by jvoisin** [comment]
Please don't bundle a snapshot of signify, since this will add burden to keeping everything up to date. We currently have enough issues with grub/libmagic/regexp/…

**Comment by radare** [comment]
All the projects i merged are dead, abandoned or stable enough to not need to be frequently updated. Signify fits in the same category.

**Comment by XVilka** [comment]
Ok, then something like a short instruction how to verify signature of the downloaded files.

**Comment by radare** [comment]
Its described in the makefile iirc

**Comment by radare** [comment]
Done by radare2-release, not signed.. but checksummed
